lib/sqlalchemy/orm/identity.py

```diff
@@ -101,7 +101,7 @@ def _manage_incoming_state(self, state: InstanceState[Any]) -> None:
         if state.modified:
             self._modified.add(state)
 
-    def _manage_removed_state(self, state: InstanceState[Any]) -> None:
+    def _manage_removed_state(self, state: InstanceState[Any] | None) -> None:
```

Comment:
```suggestion
    def _manage_removed_state(self, state: InstanceState[Any]) -> None:
```

optional should no longer be needed here